product_attribute_value_archive/tests/test_archive_attribute_value.py

```diff
@@ -75,18 +74,22 @@ def setUpClassCommon(cls):
             ]
         )
         cls.red_pen = pen_template_value.ptav_product_variant_ids
+        cls.order = cls.env.ref("sale.sale_order_1")
 
     @classmethod
-    def _create_sale_order(cls, products):
+    def _create_sale_order_lines(cls, products):
         """Create a sale order with given products."""
-        sale_form = Form(cls.env["sale.order"])
-        sale_form.partner_id = cls.partner
-        with mute_logger("odoo.tests.common.onchange"):
-            for product in products:
-                with sale_form.order_line.new() as line:
-                    line.product_id = product
-                    line.product_uom_qty = 1
-        return sale_form.save()
+        vals = []
+        for product in products:
+            vals.append(
+                {
+                    "product_id": product.id,
+                    "product_uom_qty": 1,
+                    "order_id": cls.order.id,
+                }
+            )
+        cls.env["sale.order.line"].create(vals)
+        return cls.order
```

Comment:
We should avoid impure functions, here you update a variable (`cls.order`) that is declared outside of the method, while the name of the method and its parameters suggests you are creating order lines only.
Why did you change the previous implementation?

If there is a good reason to change the implementation, I would write:
```python
def _create_sale_order_lines(cls, order, products):
    ...
    return cls.order
```
And pass explicitly the order on each call.

Reply:
The issue is that, sale_stock is installed by another module and introduces a required field on sale_order.
However, we just have the constraint on the database, but the fields doesn't exists in the odoo context.
Then, odoo doesn't set the default value on this fields, and creating sale order fails.
I've found this workaround : adding so lines to on an existing sale order.